src/tests/JIT/Methodical/Boxing/boxunbox/KeepAliveBoxOpt.cs

```diff
@@ -0,0 +1,355 @@
+using System;
+using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
+using System.Runtime.Intrinsics;
+
+public class KeepAliveBoxFieldsTest
+{
+    private static string _failedTest;
+    private static string _failedFieldName;
+
+    public static int Main()
+    {
+        TestLocalsOneBlock();
+        Reset();
+        TestImplicitByrefsOneBlock();
+        Reset();
+        TestExplicitByrefsOneBlock();
+        Reset();
+        TestClassWrapperOneBlock();
+        Reset();
+        TestFinallyLocals();
+        Reset();
+        TestThrowCatchFinallyLocals();
+        Reset();
+
+        return 100;
+    }
+
+    public static void TestLocalsOneBlock()
+    {
+        var s1 = new ComplexStructWithExplicitLayout();
+        s1.ObjectField = new Reporter("s1.ObjectField", nameof(TestLocalsOneBlock));
+        s1.AnotherObjectField = new Reporter("s1.AnotherObjectField", nameof(TestLocalsOneBlock));
+        s1.StructWithRef.ObjectField = new Reporter("s1.StructWithRef.ObjectField", nameof(TestLocalsOneBlock));
+        var s2 = new SimpleStructWithExplicitLayout();
+        s2.ObjectField = new Reporter("s2.ObjectField", nameof(TestLocalsOneBlock));
+        var s3 = new SimpleStructWithAutoLayout();
+        s3.ObjectField = new Reporter("s3.ObjectField", nameof(TestLocalsOneBlock));
+        s3.AnotherObjectField = new Reporter("s3.AnotherObjectField", nameof(TestLocalsOneBlock));
+        var s4 = new SingleObjectStruct();
+        s4.ObjectField = new Reporter("s4.ObjectField", nameof(TestLocalsOneBlock));
+        var s5 = new StructWithoutObjectFields();
+        var s6src = new SingleObjectStruct();
+        s6src.ObjectField = new Reporter("s6src.ObjectField", nameof(TestLocalsOneBlock));
+        var s6 = new SingleObjectStruct?(s6src);
+
+        GC.Collect(2, GCCollectionMode.Forced);
+        GC.WaitForPendingFinalizers();
+
+        GC.KeepAlive(s1);
```

Comment:
I think to make this test GC-stress compatible, the KeepAlive calls need to come after the CheckSuccess call.

Reply:
Yes, of course. Not sure what my thinking here was.

Reply:
I've definitely made the same kind of mistake myself. No worries!